samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/ImageGenerationAgentService.cs

```diff
@@ -0,0 +1,312 @@
+        IConfiguration configuration,
+        IHttpClientFactory httpClientFactory) 
+        : base(configuration[EndpointConfigKey] ?? 
+              throw new InvalidOperationException($"Configuration key '{EndpointConfigKey}' not set"), 
+              logger,
+              configuration)
+    {
+        _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+        
+        // Check if DALL-E endpoint is set in configuration
+        string? dalleEndpoint = configuration[DallEEndpointKey];
+        
+        // Only enable DALL-E if endpoint is provided
+        _dalleEnabled = !string.IsNullOrEmpty(dalleEndpoint);
+        
+        if (_dalleEnabled)
+        {
+            Logger.LogInformation($"DALL-E image generation is enabled with Microsoft Entra ID authentication");
+            Logger.LogInformation($"Using DALL-E endpoint: {dalleEndpoint}");
+            
+            // Verify endpoint format - DALL-E endpoint is different from Azure AI Projects endpoint
+            if (!dalleEndpoint.Contains("openai.azure.com") && !dalleEndpoint.Contains("api-version"))
+            {
+                Logger.LogWarning($"DALL-E endpoint format appears incorrect: {dalleEndpoint}");
+                Logger.LogWarning("DALL-E endpoint should follow the format: https://your-resource.openai.azure.com/openai/deployments/your-deployment-name/images/generations?api-version=2023-12-01-preview");
+                Logger.LogWarning("This is different from the Azure AI Projects endpoint used for agent creation");
+                Logger.LogWarning("Using placeholders instead of real image generation due to incorrect endpoint format");
+                
+                // Disable DALL-E if endpoint format is incorrect
+                _dalleEnabled = false;
+                _dallEEndpoint = null;
+                return;
+            }
+            
+            // Use the DALL-E endpoint from configuration directly without modification
+            // Note: The DALLE_ENDPOINT should be the complete URL including deployment name and API version
+            // Example format: https://your-resource.openai.azure.com/openai/deployments/your-deployment-name/images/generations?api-version=2023-12-01-preview
+            _dallEEndpoint = dalleEndpoint;
+            
+            // Create DefaultAzureCredential for authentication
+            _credential = new DefaultAzureCredential();
+        }
+        else
+        {
+            Logger.LogWarning("DALL-E image generation is disabled. Set DALLE_ENDPOINT environment variable to enable it.");
+            Logger.LogWarning("The DALLE_ENDPOINT should point to your Azure OpenAI service's DALL-E deployment, which is different from the AI Projects endpoint.");
+            _dallEEndpoint = null;
+        }
+    }
+    
+    /// <summary>
+    /// Initializes the agent if needed
+    /// </summary>
+    private async Task InitializeAsync()
+    {
```

Comment:
Could this be moved into the base class, and even called by the base `GetResponseAsync()` to eliminate the need for derived types to define/call it themselves?  Also, if these services were registered as singletons, then there is a potential race condition when activities executed in parallel end up calling this method concurrently.